src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -4672,6 +4682,104 @@ public void testInspectSwarm() throws Exception {
+        .advertiseAddr("127.0.0.1")
+        .listenAddr("0.0.0.0:2377")
+        .build()
+    );
+    assertThat(nodeId, is(notNullValue()));
+
+    sut.leaveSwarm(true);
+
+    // Test initializing swarm with SwarmSpec
+    final String nodeId2 = sut.initSwarm(SwarmInit.builder()
+        .advertiseAddr("127.0.0.1")
+        .listenAddr("0.0.0.0:2377")
+        .swarmSpec(SwarmSpec.builder()
+            .caConfig(CaConfig.builder().build())
+            .dispatcher(DispatcherConfig.builder().build())
+            .labels(Collections.<String, String>emptyMap())
+            .raft(RaftConfig.builder().build())
+            .encryptionConfig(EncryptionConfig.builder().autoLockManagers(true).build())
+            .taskDefaults(TaskDefaults.builder().build())
+            .build()
+        )
+        .build()
+    );
+    assertThat(nodeId2, is(notNullValue()));
+  }
+
+  @Test
+  public void testUpdateSwarm() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+
+    final Swarm swarm = sut.inspectSwarm();
+    final ThreadLocalRandom random = ThreadLocalRandom.current();
+
+    final Map<String, String> newLabels = ImmutableMap.of("foo", "bar");
+    final OrchestrationConfig newOrchestration = OrchestrationConfig.builder()
+        .taskHistoryRetentionLimit(random.nextInt(1, 10))
+        .build();
+    final RaftConfig newRaft = RaftConfig.builder()
+        .snapshotInterval(random.nextInt(1, 10000))
+        .keepOldSnapshots(random.nextInt(1, 10))
+        .logEntriesForSlowFollowers(random.nextInt(1, 1000))
+        .electionTick(random.nextInt(1, 10))
+        .heartbeatTick(random.nextInt(1, 10))
+        .build();
+    final DispatcherConfig newDispatcher = DispatcherConfig.builder()
+        .heartbeatPeriod(random.nextLong(1, 5000000000L))
+        .build();
+    final CaConfig newCa = CaConfig.builder()
+        .nodeCertExpiry(random.nextLong(1, 7776000000000000L))
+        .build();
+    final EncryptionConfig newEncryption = EncryptionConfig.builder()
+        .autoLockManagers(true)
+        .build();
+    final TaskDefaults newTaskDefaults = TaskDefaults.builder()
+        .build();
+
+    final SwarmSpec updatedSpec = SwarmSpec.builder()
+        .name(swarm.swarmSpec().name() + "2")
+        .labels(newLabels)
+        .orchestration(newOrchestration)
+        .raft(newRaft)
+        .dispatcher(newDispatcher)
+        .caConfig(newCa)
+        .encryptionConfig(newEncryption)
+        .taskDefaults(newTaskDefaults)
+        .build();
+
+    sut.updateSwarm(swarm.version().index(), updatedSpec);
+    final Swarm updatedSwarm = sut.inspectSwarm();
+
+    assertThat(updatedSwarm.id(), equalTo(swarm.id()));
+    final SwarmSpec newSpec = updatedSwarm.swarmSpec();
+    assertThat(newSpec, equalTo(updatedSpec));
+
+    // Return swarm back to old settings
+    sut.updateSwarm(updatedSwarm.version().index(), swarm.swarmSpec());
+  }
+
+  @Test
+  public void testUnlockKey() throws Exception {
```

Comment:
this seems like the perfect thing to test with a unit test. We should really be prioritizing testing how docker-client interacts with the Docker Remote API rather than the integration test of `(how docker-client interacts with Docker Remote API + how docker daemon behaves/if it implements that API)`.

for instance in a unit test you could just check that the client sends HTTP requests that are correct according to the API spec. We should really not care so much that the daemon also implements the API spec as advertised.